llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -2041,6 +2041,28 @@ void CombinerHelper::applyCombineMulToShl(MachineInstr &MI,
   Observer.changedInstr(MI);
 }
 
+bool CombinerHelper::matchCombineSubToAdd(MachineInstr &MI,
+                                          BuildFnTy &MatchInfo) {
+  assert(MI.getOpcode() == TargetOpcode::G_SUB && "Expected a G_SUB");
+  auto MaybeImmVal =
+      getIConstantVRegValWithLookThrough(MI.getOperand(2).getReg(), MRI);
```

Comment:
There are no copies. They get combined away. Depending on how you design the MIR pattern,
```
 APInt Input = getIConstantFromReg(Sub->getRHSReg(), MRI);
```
might be easier and more efficient.